modules/core/src/bitgo.ts

```diff
@@ -427,848 +699,920 @@ const BitGo = function(params) {
-
-  // calculate the SHA256 hash of the token
-  const hashDigest = sjcl.hash.sha256.hash(token);
-  const tokenHash = sjcl.codec.hex.fromBits(hashDigest);
-  return {
-    hmac,
-    timestamp,
-    tokenHash
-  };
-};
+  /**
+   *
+   */
+  preprocessAuthenticationParams({ username, password, otp, forceSMS, extensible, trust }: AuthenticateOptions): ProcessedAuthenticationOptions {
+    if (!_.isString('username')) {
+      throw new Error('expected string username');
+    }
 
-/**
- * Verify the HMAC for an HTTP response
- * @param url
- * @param statusCode
- * @param text
- * @param timestamp
- * @param token Authentication token
- * @param hmac
- * @returns {{isValid: boolean, expectedHmac: *, signatureSubject: *}}
- */
-BitGo.prototype.verifyResponse = function({ url: urlPath, statusCode, text, timestamp, token, hmac }) {
-  const signatureSubject = this.calculateHMACSubject({ urlPath, text, timestamp, statusCode });
+    if (!_.isString('password')) {
+      throw new Error('expected string password');
+    }
 
-  // calculate the HMAC
-  const expectedHmac = this.calculateHMAC(token, signatureSubject);
+    const lowerName = username.toLowerCase();
+    // Calculate the password HMAC so we don't send clear-text passwords
+    const hmacPassword = this.calculateHMAC(lowerName, password);
 
-  // verify the HMAC and timestamp
-  return {
-    isValid: expectedHmac === hmac,
-    expectedHmac,
-    signatureSubject
-  };
-};
+    const authParams: ProcessedAuthenticationOptions = {
+      email: lowerName,
+      password: hmacPassword,
+      forceSMS: !!forceSMS,
+    };
 
-/**
- *
- * @param params
- * @param callback
- */
-BitGo.prototype.preprocessAuthenticationParams = function(params) {
-  params = params || {};
-  common.validateParams(params, ['username', 'password'], ['otp']);
-
-  const username = params.username.toLowerCase();
-  const password = params.password;
-  const otp = params.otp;
-  const trust = params.trust;
-
-  // Calculate the password HMAC so we don't send clear-text passwords
-  const hmacPassword = this.calculateHMAC(username, password);
-
-  const authParams: any = {
-    email: username,
-    password: hmacPassword,
-    forceSMS: !!params.forceSMS
-  };
+    if (otp) {
+      authParams.otp = otp;
+      if (trust) {
+        authParams.trust = 1;
+      }
+    }
 
-  if (otp) {
-    authParams.otp = otp;
-    if (trust) {
-      authParams.trust = 1;
+    if (extensible) {
+      this._extensionKey = makeRandomKey();
+      authParams.extensible = true;
+      authParams.extensionAddress = this._extensionKey.getAddress();
     }
-  }
 
-  if (params.extensible) {
-    this._extensionKey = makeRandomKey();
-    authParams.extensible = true;
-    authParams.extensionAddress = this._extensionKey.getAddress();
+    return authParams;
   }
 
-  return authParams;
-};
+  /**
+   * Login to the bitgo platform.
+   */
+  authenticate(params: AuthenticateOptions, callback?: NodeCallback<any>): Bluebird<any> {
```

Comment:
looks like we can ```yield authenticate``` now right? is this a breaking change though?

Reply:
I don't think it is, as you could do `yield authenticate()` before this change as well. It returned a promise before, but now we use the coroutine syntax instead of .then()